### ISSUE TITLE
GPS Lat and Long Reference removed from EXIF data

### ISSUE THREAD

**Issue by lintujuh** [nemethviktor/GeoTagNinja#72]
**Describe the bug**
Setting geolocation to a western longitude creates conflicting EXIF information

**To Reproduce**
1. Geotag an image with western coordinates
2. Save the image
3. Validate the coordinates with exiftool

**Expected behavior**
All GPS related EXIF fields are updated correctly

**Screenshots**
```
exiftool -a -G \Users\xxx\Pictures\2023\03\2023_03_01\_3010149.ORF | findstr /i gps
[EXIF]          GPS Version ID                  : 2.3.0.0
[EXIF]          GPS Latitude Ref                : North
[EXIF]          GPS Latitude                    : 38.707384
[EXIF]          GPS Longitude Ref               : West
[EXIF]          GPS Longitude                   : 9.142500
[EXIF]          GPS Altitude Ref                : Above Sea Level
[EXIF]          GPS Altitude                    : 10.6 m
[EXIF]          GPS Speed                       : 0
[Composite]     GPS Altitude                    : 10.6 m Above Sea Level
[Composite]     GPS Latitude                    : +38.707384
[Composite]     GPS Longitude                   : -9.142500
[Composite]     GPS Position                    : +38.707384, -9.142500
```
Note the difference between composite and EXIF GPS Longitude Ref. Also if I remove the `.Exiftool_config` the Composite section looks like
```
[Composite]     GPS Altitude                    : 10.6 m Above Sea Level
[Composite]     GPS Latitude                    : 38 deg 42' 26.58" N
[Composite]     GPS Longitude                   : 9 deg 8' 33.00" W
[Composite]     GPS Position                    : 38 deg 42' 26.58" N, 9 deg 8' 33.00" W
```

**Desktop (please complete the following information):**
 - OS: Win11 Enterprise 22H2
 - Language/Culture:  en-FI

**Additional context**
An example file geotagged to Lisbon to which the output is referring to
[_3010149.zip](https://github.com/nemethviktor/GeoTagNinja/files/10939281/_3010149.zip)

**Comment by nemethviktor** [comment]
Morning, 

If you are referring to the discrepancy that 
`[EXIF]          GPS Longitude                   : 9.142500` is positive whereas 
`[Composite]     GPS Longitude                   : -9.142500` is negative, 

I think that's "normal". My recollection is that the non-composite value is to be processed in conjunction with the `ref` part and so the actual numeric value is always positive. 

I did a save/overwrite (with a very slightly modified coordinate) on your file in the old geosetter (v3) and if I use "my" config file I get the same
![image](https://user-images.githubusercontent.com/58767235/224285104-c7e493fc-7878-44a8-81b5-60ef7dc9c5e4.png)

**Comment by StarGeekSpaceNerd** [comment]
> My recollection is that the non-composite value is to be processed in conjunction with the ref part and so the actual numeric value is always positive.

This is correct.  The EXIF coordinate values are saved as set of 3 unsigned 64bit rational numbers, i.e. 3 fractions.

Example of how `9 deg 8' 33.00"` is stored in the file  
```
  | | 2)  GPSLongitude = 9 8 33 (9/1 8/1 33/1)
  | |     - Tag 0x0004 (24 bytes, rational64u[3]):
  | |         0090: 00 00 00 09 00 00 00 01 00 00 00 08 00 00 00 01 [................]
  | |         00a0: 00 00 00 21 00 00 00 01
```

**Comment by lintujuh** [comment]
I managed to reproduce the steps that create the issue.

1. Open Import GPX/JSON File dialog
2. Select attached GPX File
3. Tick use this time zone and select London (as there is no Lisbon)
4. Ignore the exception
5. Tick Do not do reverse geocoding
6. Save changes
7. Close the folder and reopen it
8. See the image has moved from Lisbon to south of Sardinia
9. Check EXIF data and see that GPS Latitude Ref and GPS Longitude Ref tags have been removed

Attached is the original RAW file and the related GPX file.

[orig_2280104.zip](https://github.com/nemethviktor/GeoTagNinja/files/10948471/orig_2280104.zip)

**Comment by nemethviktor** [comment]
I am struggling to reproduce this even in the en_fi VM -- however, can you have a look at the development version please at https://drive.google.com/file/d/18iI77SIdrIv-joOtyT0-MzqOVtB5OgM0/view?usp=share_link

This is actually using a very different internal logic to what's in the Master branch and I'm hoping it won't cause the problem you have. Let me know either way pls. It also has a fix for the TimeZone dropdown crash. I haven't yet added the Country dropdown as per other request, that's kinda next up. When that's done I'll leave a comment in the relevant thread.

**Comment by lintujuh** [comment]
The dev version seems to work.

**Comment by lintujuh** [comment]
There is still some sporadic issue. I had ~80 images to which I imported the geolocations from a gpx file. I went through the images and copy-pasted the locations from to some others so that images taken from the same place had the same coordinates. I guess I did some 30 copy-paste operations, and in two cases the images moved close to Sardinia from Lisbon. When I manually corrected the longitude from 9,something to -9,something and saved, the issue was fixed.
